.github/workflows/install-valkey/action.yml

```diff
@@ -26,8 +31,8 @@ runs:
 
     steps:
         - name: Cache Valkey
-           # TODO: remove the musl ARM64 limitation when https://github.com/actions/runner/issues/801 is resolved
-          if: ${{ inputs.target != 'aarch64-unknown-linux-musl' }}
+          # TODO: remove the musl ARM64 limitation when https://github.com/actions/runner/issues/801 is resolved
+          if: ${{ inputs.use-cache == true && inputs.target != 'aarch64-unknown-linux-musl' }}
```

Comment:
Why we need a flag to use cache? We always need to use cache to avoid re-compilation.